rllib/algorithms/callbacks.py

```diff
@@ -51,13 +52,42 @@ def __init__(self, legacy_callbacks_dict: Dict[str, callable] = None):
             )
         self.legacy_callbacks = legacy_callbacks_dict or {}
 
+    @OverrideToImplementCustomLogic
```

Comment:
resorted by order of the events usually happening: algo init first, sub-env created, etc..